src/main/java/org/lanternpowered/server/inject/option/OptionObjectProvider.java

```diff
@@ -46,92 +50,38 @@
 @SuppressWarnings("unchecked")
 abstract class OptionObjectProvider<T> implements Provider<T> {
 
-    static class IntegerImpl extends OptionObjectProvider<Integer> {
-        @Inject
-        public IntegerImpl() {
-            super(Integer.class);
-        }
-    }
-
-    static class DoubleImpl extends OptionObjectProvider<Double> {
-        @Inject
-        public DoubleImpl() {
-            super(Double.class);
-        }
-    }
-
-    static class FloatImpl extends OptionObjectProvider<Float> {
-        @Inject
-        public FloatImpl() {
-            super(Float.class);
-        }
-    }
-
-    static class LongImpl extends OptionObjectProvider<Long> {
-        @Inject
-        public LongImpl() {
-            super(Long.class);
-        }
-    }
-
-    static class ShortImpl extends OptionObjectProvider<Short> {
-        @Inject
-        public ShortImpl() {
-            super(Short.class);
-        }
-    }
-
-    static class BooleanImpl extends OptionObjectProvider<Boolean> {
-        @Inject
-        public BooleanImpl() {
-            super(Boolean.class);
-        }
-    }
-
-    static class StringImpl extends OptionObjectProvider<String> {
-        @Inject
-        public StringImpl() {
-            super(String.class);
-        }
-    }
-
-    static class ByteImpl extends OptionObjectProvider<Byte> {
-        @Inject
-        public ByteImpl() {
-            super(Byte.class);
-        }
-    }
-
-    static class FileImpl extends OptionObjectProvider<File> {
-        @Inject
-        public FileImpl() {
-            super(File.class);
-        }
-    }
-
     static class PathImpl extends OptionObjectProvider<Path> {
         @Inject
         public PathImpl() {
-            super(File.class);
         }
 
         @Override
         public Path get() {
             final File file = (File) get0();
             return file == null ? null : file.toPath();
         }
+
+        @Override
+        protected ValueConverter<Path> typeConverter() {
+            return new PathConverter();
+        }
     }
 
+    static class BooleanImpl extends OptionObjectProvider<Boolean> {}
+    static class ByteImpl extends OptionObjectProvider<Byte> {}
+    static class DoubleImpl extends OptionObjectProvider<Double> {}
+    static class FileImpl extends OptionObjectProvider<File> {}
+    static class FloatImpl extends OptionObjectProvider<Float> {}
+    static class IntegerImpl extends OptionObjectProvider<Integer> {}
+    static class LongImpl extends OptionObjectProvider<Long> {}
+    static class ShortImpl extends OptionObjectProvider<Short> {}
+    static class StringImpl extends OptionObjectProvider<String> {}
+
+    protected final TypeToken<T> type = new TypeToken<T>(this.getClass()) {};
```

Comment:
Doens't this create a `TypeToken<OptionObjectProvider<T>>`? And to get the `T` parameter you use `new TypeToken<OptionObjectProvider<T>>(getClass()).resolve(OptionObjectProvider.class.getTypeParameters()[0])`?